### PR TITLE
Anonymode

### PR DESCRIPTION
**Pull request by annirudh**

### CODE REVIEW

**Comment by annirudh** [comment]
> We want the tty check in wandb.init if api.api_key is null. This way someone can just run a script after checking out a repo. See
> 
> https://github.com/wandb/client/blob/ef59404d57be8932bfd8abe5f65b67a67b2aa6a4/wandb/__init__.py#L832
> 
> We already have a force parameter that should be respected to prevent anonymode. The jupyter integration is a little trickier, I can handle that piece.

Gotcha, I'll fix this. Thanks!

**Comment by vanpelt** [comment]
Also, I think `sys.__stdin__.isatty()` is a better check for interactive input.  We should also add an env variable to env.py that lets a user opt-in to anonymode regardless of tty.

File: wandb/wandb_run.py [hunk 1]
```diff
@@ -390,6 +391,23 @@ def save(self, id=None, program=None, summary_metrics=None, num_retries=None, ap
         self.name = upsert_result.get('displayName')
         return upsert_result
 
+    def check_anonymous(self):
+        # If there's no API key set, ask if the run should be logged anonymously. Only launch this prompt in
+        # environments with a tty.
+        if not self.api.api_key and sys.stdin.isatty():
+            resp = str(input('No API key found. Would you like to log runs anonymously? (y/n)\n').lower().strip())
```
Comment (vanpelt) [inline wandb/wandb_run.py]: We need to make this clearer it's wandb asking for input.  Ideally we would prefix it with the blue `wandb:` and print the app_url in this message, i.e. log runs anonymously to https://app.wandb.ai

Comment (annirudh) [reply wandb/wandb_run.py]: Screen Shot 2019-07-30 at 3 29 19 PM

How does this look?

Comment (vanpelt) [reply wandb/wandb_run.py]: Looks good, one minor nit would be to make the Appending key message have wandb: as well, but that's been like that forever.

**Pull request merged**